WikiWatershed/model-my-watershed
# Various Chart Improvements

The following changes are presented:

## Legend Improvements ##

### Legend Clicking ###

It is now possible to disable clickability of the legend by setting disableToggle to true in the options hash sent to the vertical chart maker.

Connects #1022

### Legend Order ###

It is now possible to reverse the order in which items appear in the legend.

Connects #1023

## Vertical Chart Label Staggering ##

Make Staggering Dependant on Chart Width
    
Connects #1021

## Testing Instructions ##

   * Pull down this `nvd3` branch: https://github.com/azavea/nvd3/pull/1
   * "Build" and "install" it with something like the following command `cat src/*.js src/models/*.js > ~/local/src/model-my-watershed/src/mmw/node_modules/nvd3/build/nv.d3.js`
   * In MMW, rebuild the vendor dependencies: `./scripts/bundle --debug --vendor`
   * Use the application:  Everything should be as it was before, except that items in the legends of the charts in the compare view are in reverse order and are not clickable.  It should now also be possible to observe that the x-axis labels of horizontal charts are staggered when those charts of of width 420 pixels or less, other wise they are not.
